### PR TITLE
set content_length as len(body) if response ContentLength is -1

### PR DESCRIPTION
**Pull request by parrasajad**
## Checklist

- [x] Pull request is created against the [dev](https://github.com/projectdiscovery/nuclei/tree/dev) branch
- [x] All checks passed (lint, unit/integration/regression tests etc.) with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: v2/pkg/protocols/http/operators.go [hunk 1]
```diff
@@ -121,7 +121,12 @@ func (request *Request) responseToDSLMap(resp *http.Response, host, matched, raw
 	data["response"] = rawResp
 	data["status_code"] = resp.StatusCode
 	data["body"] = body
-	data["content_length"] = resp.ContentLength
+	if resp.ContentLength > -1 {
```
Comment (Mzack9999) [inline v2/pkg/protocols/http/operators.go:124]: Can we pack this into a standard helper function that calculates the correct int64 length, for example, at `github.com/projectdiscovery/nuclei/v2/pkg/protocols/http/utils.go`?

**Review by Mzack9999 — CHANGES_REQUESTED** [review]
Please see my comment, and let's add test cases to cover the scenarios mentioned in the issue description.

**Review by Mzack9999 — CHANGES_REQUESTED** [review]
I'd recommend adding an integration test (e.g., for the `http` protocol) with a couple of requests that ensure the correct behavior for the `content_length` calculation in case the server returns the content length or the header is absent.

**Review by Mzack9999 — APPROVED** [review]

**Review by ehsandeep — APPROVED** [review]
```console
echo https://m.facebook.com | ./nuclei -t test.yaml -v

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   2.7.8

		projectdiscovery.io

[WRN] Use with caution. You are responsible for your actions.
[WRN] Developers assume no liability and are not responsible for any misuse or damage.
[INF] Using Nuclei Engine 2.7.8 (latest)
[INF] Using Nuclei Templates 9.2.5 (latest)
[INF] Templates added in last update: 40
[INF] Templates loaded for scan: 1
[VER] [basic-example] Sent HTTP request to https://m.facebook.com
[2022-10-24 16:35:14] [basic-example] [http] [info] https://m.facebook.com [53888]
```

**Pull request merged**